DevEdu.API/DevEdu.Business/Services/CourseService.cs

```diff
@@ -44,5 +47,49 @@ public List<CourseTopicDto> SelectAllTopicsByCourseId(int courseId)
             var list = _courseRepository.SelectAllTopicsByCourseId(courseId);
             return list;
         }
+        public void UpdateCourseTopicsByCourseId(int courseId, List<CourseTopicDto> topics)
+        {
+            CheckUniquenessPositions(topics);
+            var topicsInDatabase = _courseRepository.SelectAllTopicsByCourseId(courseId);
+            if (((topics.Count > topicsInDatabase.Count) || (topics.Count < topicsInDatabase.Count)) && topicsInDatabase.Count!=0)
+            {
+                DeleteAllTopicsByCourseId(courseId);
+                AddTopicsFromNewList(courseId, topics);
+            }
+            else if( topicsInDatabase.Count == 0)
+            {
+                AddTopicsFromNewList(courseId, topics);
+            }
+            else
+            {
+                foreach (var topic in topics)
+                {
+                    topic.Course = SetIdToCourseDto(courseId, topic.Course);
+                }
+                _courseRepository.UpdateCourseTopicsByCourseId(courseId, topics);
+            }
+        }
+        public void DeleteAllTopicsByCourseId(int courseId)
+        {
+            _courseRepository.DeleteAllTopicsByCourseId(courseId);
+        }
+        private void AddTopicsFromNewList(int courseId, List<CourseTopicDto> topics)
+        {
+            foreach (var topic in topics)
+            {
+                AddTopicToCourse(courseId, topic.Topic.Id, topic);
```

Comment:
we might have ~50 topics in a course
and it means that we'll send 50 requests to DB here
I think it's not the best solution(

Reply:
fixed